python/ray/serve/tests/test_target_capacity.py

```diff
@@ -373,6 +380,358 @@ def test_autoscaling_scale_to_zero(
     )
 
 
+@serve.deployment(ray_actor_options={"num_cpus": 0})
+class ControlledLifecycleDeployment:
+    async def __init__(self):
+        self.lifecycle_signal = ray.get_actor(
+            name="lifecycle_signal", namespace=SERVE_NAMESPACE
+        )
+        await self.lifecycle_signal.wait.remote()
+
+        self.request_signal = ray.get_actor(
+            name="request_signal", namespace=SERVE_NAMESPACE
+        )
+
+    async def __call__(self) -> str:
+        try:
+            await self.request_signal.wait.remote()
+        except RayActorError:
+            print("Request signal actor already dead. Running request.")
+        return "Hello world!"
+
+    async def __del__(self):
+        # Use try-except here to avoid cluttering logs if the SignalActor
+        # gets torn down before the replicas during clean up.
+        try:
+            await self.lifecycle_signal.wait.remote()
+        except RayActorError:
+            print("Lifecycle signal actor already dead. Replica exiting.")
+
+
+def create_controlled_app(config: Dict) -> Application:
```

Comment:
Better to use pydantic-typed `config` here :)

Reply:
Sounds good, I used a pydantic-typed `config`.